### PR TITLE
LSIF: extract common DB columns into lsif_dumps table

### PR DESCRIPTION
**Pull request by chrismwendt**
**Impact** This is purely a refactor of DB tables in preparation for multi-lang/project support. There's no real logic or behavior being changed here.

**Motivation** Multi-lang/project support will require a set of wildcards to be associated with each LSIF dump, and doing so on the DB schema prior to this PR would result in a `wildcards` column attached to a few different tables, resulting in duplication of data and an increased chance of bugs. After this PR, we'll only need to add the new column `wildcards` to one table.

**Implementation**

- Factors out the `repository` and `commit` columns from a few tables and puts them into a dedicated `lsif_dumps` table
- Drops the `lsif_data_markers` table (superseded by `lsif_dumps`)

See the [RFC discussion](https://docs.google.com/a/sourcegraph.com/document/d/1ocoxE_Jnee70_RjQEFSz58jxRE3VZuNg9OGHvu68O6o/edit?disco=AAAADd4IacA)

Test plan: manually tried it on https://github.com/chrismwendt/cross-app and https://github.com/chrismwendt/cross-lib

### CODE REVIEW

File: lsif/src/commits.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export function mod(sum: string, max: number): number {
  * @param commit The commit from which the gitserver queries should start.
  */
 async function getCommitsNear(gitserverUrl: string, repository: string, commit: string): Promise<[string, string][]> {
-    const args = ['git', 'log', '--pretty=%H %P', commit, `-${MAX_COMMITS_PER_UPDATE}`]
+    const args = ['log', '--pretty=%H %P', commit, `-${MAX_COMMITS_PER_UPDATE}`]
```
Comment (efritz) [inline lsif/src/commits.ts]: This snuck in from your base branch.

File: lsif/src/xrepo.ts [hunk 2]
```diff
@@ -229,49 +229,36 @@ export class XrepoDatabase {
...
-            await entityManager
-                .createQueryBuilder()
-                .delete()
-                .from(ReferenceModel)
-                .where({ repository, commit })
-                .execute()
+            // Mark that we have data available for this commit
+            const result = await entityManager.createQueryBuilder().insert().onConflict('DO NOTHING').into(LsifDump).values({repository, commit}).execute()
+            if (result.identifiers.length === 0) {
+                throw new Error(`Unable to insert row into lsif_dumps table for repository ${repository} commit ${commit}. Make sure migrations have run.`)
```
Comment (efritz) [inline lsif/src/xrepo.ts]: Migrations must have run to get this far into the app.

File: lsif/src/logging.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ function toLogLevel(value: string): 'debug' | 'info' | 'warn' | 'error' {
 /**
  * The maximum level log message to output.
  */
-const LOG_LEVEL: 'debug' | 'info' | 'warn' | 'error' = toLogLevel((process.env.LOG_LEVEL || 'info').toLowerCase())
+const LOG_LEVEL: 'debug' | 'info' | 'warn' | 'error' = toLogLevel((process.env.LOG_LEVEL || 'debug').toLowerCase())
```
Comment (efritz) [inline lsif/src/logging.ts]: Why this change? Debug logs are very noisy.

Comment (chrismwendt) [reply lsif/src/logging.ts]: Oops, didn't mean to change the log level.

**Comment by chrismwendt** [comment]
Thanks, @efritz, I addressed your comments.

**Comment by efritz** [comment]
I merged with master and updated the names of the tables after merging #5935.

File: migrations/1528395601_lsif_dumps.down.sql [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        c.repository,
+        c."commit",
+        c.parent_commit,
+        EXISTS (
+            SELECT 1
+            FROM lsif_data_markers m
+            WHERE m.repository = c.repository
+            AND m."commit" = c."commit"
+        ) AS has_lsif_data
+    FROM commits c;
```
Comment (efritz) [inline migrations/1528395601_lsif_dumps.down.sql]: lsif_commits

Comment (chrismwendt) [reply migrations/1528395601_lsif_dumps.down.sql]: Thanks, fixed.
